add rvv vpx_comp_avg_pred

sunmin89 opened the pull request:
The optimization logic is taken from corresponding neon code
 Test entry was added to comp_avg_pred_test.cc

Test methods:

1.compile
CROSS=/home/summer/nvme/bins/gnu-toolchain-bins-rvv-1.0/bin/riscv64-unknown-linux-gnu- ../configure --target=riscv64-linux-gcc --enable-unit-tests --enable-debug --disable-optimizations

2. run
qemu-riscv64 -cpu rv64,v=true,vlen=1024,vext_spec=v1.0 -L $RISCV/sysroot ./test_libvpx --gtest_filter='RVV/AvgPredTestLBD*'

3.debug
qemu-riscv64 -cpu rv64,v=true,vlen=1024,vext_spec=v1.0 -g 1234 -L $RISCV/sysroot ./test_libvpx --gtest_filter='RVV/AvgPredTestLBD*'

riscv64-unknown-linux-gnu-gdb ./test_libvpx -ex "target remote localhost:1234"

Commit:
add rvv vpx_comp_avg_pred

 The optimization logic is taken from corresponding neon code
 Test entry was added to comp_avg_pred_test.cc

unicornx commented on test/comp_avg_pred_test.cc:
```diff
@@ -227,6 +227,11 @@ INSTANTIATE_TEST_SUITE_P(NEON, AvgPredTestLBD,
+#if HAVE_RVV
+INSTANTIATE_TEST_SUITE_P(RVV, AvgPredTestLBD,
+                         ::testing::Values(&vpx_comp_avg_pred_rvv));
+#endif  // HAVE_AVX2
```
`HAVE_AVX2` -> `HAVE_RVV`

unicornx commented:
我先提几个 general 的 comments
- coding style 的问题，我对比了一下libvpx 的编码风格，我建议你检查一下所有的 c 关键字，譬如 `if`, `do`, `while`, 这些单词的 **前后** 都要加空格和其他字符隔开
- 请说明一下测试优化的效果前后对比，建议写在 commit 的 comment 中。可以包括：
  - 编译后的指令和 c 方式编译后指令数的比较
  - 如果有测试环境，我估计是 qemu，那么测试的效果在速度上优化程度如何？

另外我有一个 question：dsp 部分的函数是用来干什么的，我当时没有看懂，你看明白了吗？

Trumeet committed:
RISC-V: optimize vp8_copy_mem with RVV

Test environment: 8c 1804Mhz i5-1140G7

RVV Impl:

% CROSS=riscv64-unknown-linux-gnu- configure --target=riscv64-linux-gcc \
	--enable-debug --enable-gprof && make -j
% time qemu-riscv64 -cpu rv64,v=true,zba=true,vlen=128 -L /path/to/sysroot/ \
	./vpxenc --codec=vp8 -w 352 -h 288 -o akiyol.vpx ./akiyo_cif.yuv
Pass 1/1 frame  300/300   314977B    8399b/f  251981b/s   92226 ms (3.25 fps)
user    1m30.108s
% gprof -abp ./vpxenc ./gmon.out | grep vp8_copy_mem
  1.36     53.09     1.04  1025863     0.00     0.00  vp8_copy_mem16x16_rvv
  0.72     59.01     0.55  1641368     0.00     0.00  vp8_copy_mem8x8_rvv
  0.05     65.95     0.04   764377     0.00     0.00  vp8_copy_mem8x4_rvv

C Impl:

% CROSS=riscv64-unknown-linux-gnu- configure --target=generic-gnu --enable-debug \
	--enable-gprof && make -j
% time qemu-riscv64 -cpu rv64,v=true,zba=true,vlen=128 -L /path/to/sysroot/ \
	./vpxenc --codec=vp8 -w 352 -h 288 -o akiyol.vpx ./akiyo_cif.yuv
Pass 1/1 frame  300/300   314977B    8399b/f  251981b/s   98417 ms (3.05 fps)
user    1m36.146s
% gprof -abp ./vpxenc ./gmon.out | grep vp8_copy_mem
  0.38     63.96     0.31                             vp8_copy_mem8x4_c
  0.04     70.61     0.03   204336     0.00     0.00  vp8_copy_mem16x16_c

Signed-off-by: Yuuta Liang <yuuta@yuuta.moe>

unicornx commented on vpx_dsp/riscv/avg_pred_rvv.c:
```diff
@@ -0,0 +1,120 @@
+
+void vpx_comp_avg_pred_rvv(uint8_t *comp_pred, const uint8_t *pred, int width,
+                         int height, const uint8_t *ref, int ref_stride) {
+if(width > 8){
```
block 里面的代码行怎么不缩进的呢？
